ply-plutarch/src/Ply/Plutarch/TypedWriter.hs

```diff
@@ -75,13 +78,19 @@ For a description of 'ScriptVersion' is determined, see: 'VersionOf' type family
 
 For a description of 'ScriptRole' is determined, see: 'RoleOf' type family.
 -}
-typedWriterInfo ::
+mkEnvelope ::
   forall ptype.
   TypedWriter ptype =>
   Config ->
+  Text ->
   ClosedTerm ptype ->
-  (ScriptVersion, ScriptRole, [Typename], Either Text Script)
-typedWriterInfo conf pterm = (ver, rl, paramTypes, scrpt)
+  Either Text TypedScriptEnvelope
+mkEnvelope conf descr pterm =
```

Comment:
I think it's useful to keep a lightweight function like `typedWriterInfo` and use that in tests, and have `mkEnvelope` wrap around it.

Reply:
`typedWriterInfo` would never be used in real usages. I don't see to point of having extra interface that does things in less concise way.